modules/git/attribute/checker.go

```diff
@@ -39,6 +39,8 @@ func checkAttrCommand(gitRepo *git.Repository, treeish string, filenames, attrib
 			)
 			cancel = deleteTemporaryFile
 		}
+	} else if cached {
+		cmd.AddArguments("--cached")
```

Comment:
Please add some comments about why `cached` is used and since there is a regression, maybe we need some tests to make sure it won't break again.